hashicorp/consul
# Cc 7145 hcp link status api

### Description

https://hashicorp.atlassian.net/browse/CC-7145
Due to the API specific, it doesn't fall well to the data loading approach which is used for regular records.
So I did similar to autopilot loading, which as well not really fall under record architecture(list + crud). 

for 404 link/global and for STATE_FALSE repository maps the `isLinking` value to false. For STATE_TRUE - to true. For other error codes, repo return null for the result object (to make it possible to understand when some other system/auth error raised)



### Testing & Reproduction steps

0. add `Linked is: {{hcpLink.data.isLinked}}` in [a next line](https://github.com/hashicorp/consul/blob/2fbc2108e153a2e7271d3a552f5619bf3879b2d6/ui/packages/consul-ui/app/templates/application.hbs#L101)
1. using consul-setup repo and its [README](https://github.com/LevelbossMike/consul-setup/blob/0078b916e063196f5c130fa89497f48323207e5c/setups/peering/README.md)
2. Build consul 
3. login with ACLS `consul acl bootstrap`, using secretId
4. check the label 

It is also possible to check it with `yarn start`. Mock data set to STATE_TRUE by default

link/global could be added with triggering from browser console
```
fetch("https://localhost:8501/api/hcp/v2/link/global", {
  "headers": {
    "accept": "*/*",
    "accept-language": "en-US,en;q=0.9",
    "content-type": "application/json; charset=utf-8",
    "sec-ch-ua": "\"Not_A Brand\";v=\"8\", \"Chromium\";v=\"120\"",
    "sec-ch-ua-mobile": "?0",
    "sec-ch-ua-platform": "\"macOS\"",
    "sec-fetch-dest": "empty",
    "sec-fetch-mode": "cors",
    "sec-fetch-site": "same-origin",
    "x-consul-token": "<PAST HERE Secret ID>",
    "x-requested-with": "XMLHttpRequest"
  },
  "referrer": "https://localhost:8501/ui/dc1/services",
  "referrerPolicy": "strict-origin-when-cross-origin",
  "body": JSON.stringify(
      { data: {
          client_id: "Sq2HHIo5TjBvasTXTJW8L9yRzwjEaERs",
          resource_id: "organization/7309cdc4-d05a-4d84-af5d-40bf5adbb424/project/e38aa34e-1120-43f1-9092-c529e8595e3f/hashicorp.consul.global-network-manager.cluster/eeeeeee",
          client_secret: "elg9GrWkz6KjWBR6tgQkrix41LKcgHlUHxwArQuUZk46JFBq6KXsRhuEt8T5qyC_"
      }
      }
  ),
  "method": "PUT",
  "mode": "cors",
  "credentials": "include"
});
```


### Links



### PR Checklist
